modules/codegen-cli/src/smithy4s/codegen/cli/Main.scala

```diff
@@ -36,27 +39,55 @@ object Main {
     )
 
   def main(args: Array[String]): Unit = {
+    val argsArray = args
     val out = System.out
     try {
-      System.setOut(System.err)
       commands
         .parse(args.toList)
         .map {
           case Smithy4sCommand.Generate(args) =>
-            val res = Codegen.generateToDisk(args)
-            if (res.isEmpty) {
-              // Printing to stderr because we print generated files path to stdout
-              Console.err.println(
-                List(
-                  "Nothing was generated. Make sure your targetting Smithy files or folders",
-                  "that include Smithy definitions. Otherwise, you can also use",
-                  "--dependencies to pull external JARs or use --local-jars to use",
-                  "JARs located on your file system."
-                ).mkString(" ")
+            Console.err.println("fork value: " + args.fork)
+            if (args.fork) {
+              val cp = resolveDependencies(
+                s"com.disneystreaming.smithy4s:smithy4s-codegen-cli_2.13:${BuildInfo.version}" ::
+                  args.dependencies,
+                args.localJars,
+                args.repositories
```

Comment:
Very clever :+1: 

There is one downside tough:
This is not ideal as coursier's resolution will just pick the latest versions preventing users from using anything older than what codegen was compiled against. 

But maybe this is good enough trade-off given its low complexity? :thinking: